generator/integration_tests/benchmarks/client_benchmark.cc

```diff
@@ -0,0 +1,244 @@
+      google::test::admin::database::v1::Request const&) override {
+    return nullptr;
+  }
+
+  std::unique_ptr<::google::cloud::internal::AsyncStreamingWriteRpc<
+      google::test::admin::database::v1::Request,
+      google::test::admin::database::v1::Response>>
+  AsyncStreamingWrite(google::cloud::CompletionQueue const&,
+                      std::unique_ptr<grpc::ClientContext>) override {
+    return nullptr;
+  }
+};
+
+std::shared_ptr<GoldenKitchenSinkConnection> MakeTestConnection(
+    std::shared_ptr<GoldenKitchenSinkStub> stub, Options options) {
+  options = GoldenKitchenSinkDefaultOptions(std::move(options));
+  auto background = internal::MakeBackgroundThreadsFactory(options)();
+  return std::make_shared<GoldenKitchenSinkConnectionImpl>(
+      std::move(background), std::move(stub), std::move(options));
+}
+
+void BM_ClientRoundTripStubOnly(benchmark::State& state) {
+  auto options = Options{};
+  auto stub = std::make_shared<TestStub>();
+  auto conn = MakeTestConnection(std::move(stub), std::move(options));
+  auto client = GoldenKitchenSinkClient(std::move(conn));
+
+  for (auto _ : state) {
+    auto status = client.DoNothing();
+    benchmark::DoNotOptimize(status);
+  }
+}
+BENCHMARK(BM_ClientRoundTripStubOnly);
+
+void BM_ClientRoundTripMetadata(benchmark::State& state) {
+  auto options = Options{};
+  std::shared_ptr<GoldenKitchenSinkStub> stub = std::make_shared<TestStub>();
+  stub = std::make_shared<GoldenKitchenSinkMetadata>(std::move(stub));
+  auto conn = MakeTestConnection(std::move(stub), std::move(options));
+  auto client = GoldenKitchenSinkClient(std::move(conn));
+
+  for (auto _ : state) {
+    auto status = client.DoNothing();
+    benchmark::DoNotOptimize(status);
+  }
+}
+BENCHMARK(BM_ClientRoundTripMetadata);
+
+void BM_ClientRoundTripLogging(benchmark::State& state) {
+  auto options = Options{};
+  std::shared_ptr<GoldenKitchenSinkStub> stub = std::make_shared<TestStub>();
+  stub = std::make_shared<GoldenKitchenSinkLogging>(
+      std::move(stub), TracingOptions{}, std::set<std::string>{"rpc"});
+  auto conn = MakeTestConnection(std::move(stub), std::move(options));
+  auto client = GoldenKitchenSinkClient(std::move(conn));
+
+  for (auto _ : state) {
+    auto status = client.DoNothing();
+    benchmark::DoNotOptimize(status);
+  }
+}
+BENCHMARK(BM_ClientRoundTripLogging);
+
+void BM_ClientRoundTripTenExtraOptions(benchmark::State& state) {
+  // clang-format off
+  struct ExtraOption0 { using Type = int; };
+  struct ExtraOption1 { using Type = int; };
+  struct ExtraOption2 { using Type = int; };
+  struct ExtraOption3 { using Type = int; };
+  struct ExtraOption4 { using Type = int; };
+  struct ExtraOption5 { using Type = int; };
+  struct ExtraOption6 { using Type = int; };
+  struct ExtraOption7 { using Type = int; };
+  struct ExtraOption8 { using Type = int; };
+  struct ExtraOption9 { using Type = int; };
+  // clang-format on
+  auto options = Options{}
```

Comment:
A lot of the options we do use in every call are retry policy and backoff policy.  Those are more expensive to copy than an `int`?

Reply:
Ack. The options are defaulted in `MakeTestConnection()`, so the policies are included in each of these benchmarks..